### ISSUE TITLE
How to use multiple thread blocks for nccl AllReduce and AllGather collectives?

### ISSUE THREAD

**Issue by renganxu** [NVIDIA/nccl#657]
I am running a PyTorch model which has many SyncBatchNorm operators. The SyncBatchNorm calls nccl AllGather and AllReduce collectives. But in the kernel launch setup, the grid size is always 1, and the thread block size is either 64, 128 or 256. How to use more than one thread block (grid size > 1)? 

The model is running on 8 V100-SXM2 GPUs within one node. I tried to set NCCL_MIN_NCHANNELS to some larger values like 4 or 8, but the grid size is still 1. Because SyncBatchNorm is the main bottleneck in my model, I would like to use more GPU SMs to reduce the overhead. So how can we use multiple thread blocks for those NCCL collectives?

**Comment by sjeaugey** [comment]
This is surprising. How do you know the NCCL grid size is only 1?

**Comment by renganxu** [comment]
@sjeaugey I used the pytorch's tracing tool Kineto. The tracing tool can show the grid and block values and the time for each GPU kernel.

**Comment by renganxu** [comment]
Yes it looks like so. I also suspected that was the reason, so I checked the buffer size in each process for AllGather operation, and found that most of the time the buffer size is less than 577. When the size is 577, nccl uses grid(1, 1, 1) and block(256, 1, 1) for ncclKernel_AllGather_RING_LL_Sum_int8_t kernel. When I increased the buffer size to 4097, nccl used grid(4, 1, 1) and block(512, 1, 1). So indeed my data is a bit small.
